src/SimulationRunner.hh

```diff
@@ -491,6 +491,10 @@ namespace ignition
       /// \brief Map from file paths to Fuel URIs.
       private: std::unordered_map<std::string, std::string> fuelUriMap;
 
+      /// \brief True if Server::RunOnce triggered a blocking paused step
+      private: bool blockingPausedStepPending{false};
+
+      friend class Server;
```

Comment:
Would it be possible to add accessor functions for the private variable instead of making the server a friend? You know, social distancing the classes :wink:

Reply:
I considered adding accessor functions to the `SimulationRunner` class while I was developing this PR, however I'm not convinced enough that having setters / getters of a private property that is just an implementation detail would provide any added value. Such methods would become part of the public APIs of the `SimulationRunner` class, with all the consequences of dealing in the future with them. Would any downstream user be interested in this new variable? Unfriending a class in case of a better approach would not require any deprecation process, contrarily than removing public APIs.

If you think that these two new methods would not pollute the class and give me green light, it's a trivial change from my side for adding the following accessors:

```cpp
/// \brief Get whether the next step is going to be executed as paused.
/// \return True if the next step is being executed as paused, false otherwise.
bool NextStepIsBlockingPaused() const;

/// \brief Set the next step to be blocking and paused.
void SetNextStepAsBlockingPaused(const bool value);
```

Reply:
> Such methods would become part of the public APIs 

I appreciate the concern, but note that the `SimulationRunner`'s header is not installed, so we can modify it as needed. My personal opinion is that with the accessors we can keep better control of what parts of the `SimulationRunner` are exposed to the `Server`.

Reply:
> `SimulationRunner`'s header is not installed

Ow thanks for pointing it out, I didn't notice it. Done in ff6701b.